File: frontend/src/dtos/userDto.ts
import type { SubscriptionDto } from './subscriptionDto'
import type { UserSettingsDto } from './userSettings'

export interface UserDto {
  id: string
  fullName: string
  email: string
  settings: UserSettingsDto
  emailConfirmed: boolean
  stripeId?: string | null
  paymentType?: string | null
  paymentLastFour?: string | null
  subscriptions: SubscriptionDto[]
}

File: backend/ResQueue/ResQueue/Dtos/User/UserDto.cs
namespace ResQueue.Dtos;

public class UserDto
{
    public string Id { get; set; } = null!;
    public string FullName { get; set; } = null!;
    public string Email { get; set; } = null!;
    public bool EmailConfirmed { get; set; }

    public string Avatar { get; set; } = null!;
    public string? StripeId { get; set; }
    public string? PaymentType { get; set; }
    public string? PaymentLastFour { get; set; }

    public SubscriptionDto? Subscription { get; set; }

    public UserSettingsDto Settings { get; set; } = null!;
}